Update pkg_manager.erb to include support/detection for AmazonLinux 2 and AmazonLinux 2022

imphocused opened the pull request:
This provides detection for AmazonLinux 2022 (dnf) and fallback 'yum' support for Amazon Linux 2, as well as other RHEL'ish clones which use /etc/system-release.


<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

ekohl commented on app/views/unattended/provisioning_templates/snippet/pkg_manager.erb:
```diff
@@ -31,6 +31,10 @@ elif [ -f /etc/redhat-release ] ; then
   fi
+elif [ -f /etc/amazon-linux-release ]; then
+  PKG_MANAGER='dnf'
+elif [ -f /etc/system-release ]; then
```
Out of interest: which other systems use `/etc/system-release`? And we already source `/etc/os-release` if it's present. Can't we rely on those variables?

imphocused replied:
/etc/os-release is available, but the snippet doesn't  recognize yum in Amazon Linux 2. 

The current RHEL clones support /etc/system-release (alma,rocky,amzn). Fedora and RHEL obviously but those already have explicit support/detection.

I found the reference from this post on Redhat, [Proper way to extract Product, Release, Update, etc](https://access.redhat.com/discussions/2798501) which discusses OS detection

_3) System Release, vendor string: /etc/system-release (also via symlink, or legacy, /etc/fedora-release, /etc/redhat-release)_

The response links to a [sysconfcollect module](url) that identifies some 50+ distros near the bottom of the page. I didnt check any of them because I don't use them, though I don't expect support for them in Foreman/Katello, as I don't imagine them seeing much use (if any) in Foreman installations.

stejskalleos replied:
If we detect `/etc/system-release` can we use `dnf` instead of the `yum`?
Tested on Rocky & Alma and they both have `dnf`.

ekohl replied:
Amazon Linux 2 still has yum since it's EL7 based. Amazon Linux 2022 is based on Fedora so it has dnf.

ekohl reviewed (commented):
Could you also update the comment that starts on line 16?

stejskalleos reviewed (changes requested):
Can you update the commit message? _Update pkg_manager.erb_ is just too generic and doesn't say much about the actual changes.

stejskalleos commented:
@ekohl Changes LGTM, do you have any other comments?

ekohl commented:
It still needs a Redmine issue and the commit should refer to it. Otherwise we can't merge it (our changelog generation relies on it). Other than that I'd be happy to accept it.

stejskalleos reviewed (changes requested):
@imphocused I see that you changed the title in github but the commit message is still same: `Update pkg_manager.erb`

Commit message should be in format `Fixes #35670 - Description of changes`, where the number is id of the issue in our tracker system.

I created the [#35670 issue](https://projects.theforeman.org/issues/35670) for you, so please just update the message, for example like this: `Fixes #35670 - pkg_manager.erb - support AmazonLinux 2 & 2022`

imphocused committed:
Fixes #35670 - pkg_manager.erb - support AmazonLinux 2 & 2022

This provides detection for AmazonLinux 2022 (dnf) and fallback support for Amazon Linux 2 (yum), as well as other RHEL'ish clones

imphocused commented:
I've updated and pushed the change, sorry for the delay.

stejskalleos commented:
ok to test

stejskalleos commented:
LGTM from my side, @ekohl any other comments?

ekohl reviewed (changes requested):
I really hate it when comments are out of sync with the real code so I'd still appreciate it if the comment was updated (or removed):
https://github.com/theforeman/foreman/blob/b323ceca04e37b3958ffa3fc60c098f78098cbfd/app/views/unattended/provisioning_templates/snippet/pkg_manager.erb#L16-L24

stejskalleos commented:
@imphocused any updates? Updating comment is last thing before we can merge

imphocused committed:
Removed comments to prevent being out of sync with code, avoid listing individual distros

stejskalleos reviewed (approved):
:green_apple: **LGTM**

stejskalleos commented:
@ekohl any other comments? There are two commits now but I'll squash them before the merge
